mmcv/image/geometric.py

```diff
@@ -311,29 +313,70 @@ def imcrop(img, bboxes, scale=1.0, pad_fill=None):
         return patches
 
 
-def impad(img, shape, pad_val=0):
-    """Pad an image to a certain shape.
+def impad(img, padding, pad_val=0, padding_mode='constant'):
```

Comment:
We need to consider the backward compatibility, i.e., making codes that used old APIs work after upgrading to the latest mmcv. We can define the api like this and raise a warning is `shape` is specified.
```python
def impad(img, shape=None, padding=None, pad_val=0, padding_mode='constant'):
```
In this way we need to use keyword arguments whenever calling `impad`.